augurpy/estimators.py

```diff
@@ -1,25 +1,63 @@
-from typing import Dict, List, Optional, Union
+from dataclasses import dataclass
+from typing import Dict, Literal, Union
 
 from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
 from sklearn.linear_model import LogisticRegression
+from switchlang import switch
+from typing_extensions import TypedDict
+
+
+@dataclass
+class Params(TypedDict, total=False):
+    """Type signature for random forest and logistic regression parameters."""
+
+    n_estimators: int
+    max_depth: int
+    max_features: Union[Literal["auto"], Literal["log2"], Literal["sqrt"], int, float]
+    penalty: Union[Literal["l1"], Literal["l2"], Literal["elasticnet"], Literal["none"]]
+
+
+def _raise_exception():
+    """Raise exception for invalid classifier input."""
+    raise Exception("Missing valid input. Choose rf or lr for categorical labels and rf for continous labels.")
 
 
 def create_estimator(
-    classifier: Union[RandomForestClassifier, RandomForestRegressor, LogisticRegression],
-    rf_params: Optional[List],
-    lr_params: Optional[List],
+    classifier: Union[Literal["rf"], Literal["lr"]],
+    params: Params,
+    classification: bool = True,
 ) -> Union[RandomForestClassifier, RandomForestRegressor, LogisticRegression]:
     """Estimator object.
 
     Args:
         classifier: classifier to use in calculating the area under the curve either random forest or logistic regression
-        rf_params: list of parameters for random forest
-        lr_params: list of parameters for logistic regression
+        params: list of parameters for estimator
 
     Returns:
         Estimator object that classifies.
     """
-    pass
+    definition = classifier + " categorical: " + str(classification)
```

Comment:
definition is kind of an ambiguous name, no? If I read this without context I do not know what this is supposed to mean. Can we find a better name?

Reply:
Also, can we make this an f string?

Reply:
Hurray name finding. How about estimator_string? define_classifier? define_estimator? match_estimator? switch_estimator?
I think I like match estimator most, but switch_estimator might be less confusing🤷‍♀️

Reply:
I think we should implement it differently. Simply include more options for the `classifier` parameter: Can be one of `random_forest_regressor`, `random_forest_classifier`, [...].
Further, the classification parameter is currently not documented :)
Could you also expand a little bit on the 1 sentence description of the docstring of this method?

Something more along the lines of "Creates a model object of the provided type and populates it with desired parameters".